### DISCUSSION TITLE
FPDF2 in Anaconda Problem

### DISCUSSION THREAD

**Discussion by portfedh** [py-pdf/fpdf2#235] [General]
Hi! Im not sure this is the right place to post this, but I tried Stack Overflow and dont seem to be getting very far. 

Im having a problem using FPDF2. 

When I try to import the package, I get the following error:

**Input:**

`from fpdf import FPDF`

Response:

```
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
/var/folders/q3/38dtckns13b368s1x7g59hh40000gn/T/ipykernel_917/4051462934.py in <module>
----> 1 from fpdf import FPDF

ImportError: cannot import name 'FPDF' from 'fpdf' (unknown location)
```

I have uninstalled and reinstalled FPDF2 several times, both from PyPI pip install fpdf2 and from anaconda conda install -c conda-forge fpdf2.

Installation is successfull but code keeps running with errors. 

I'm using Python 3.8.5, conda 4.10.3 and FPDF2 2.3.1

I can get it to work if I install fpdf instead of FPDF2, but since its an older version lots of things don't work (like UTF encoding). 

Any help would be appreciated!

**Comment by Lucas-C** [comment]
Thanks for sharing your issue here.

Seems like the latest version in Anaconda is 2.3.1: https://anaconda.org/conda-forge/fpdf2/

Whereas the latest version released on Pypi is the v2.4.3: https://pypi.org/project/fpdf2/#history

You have the same error when you try to install `fpdf2` through `pip`?

For each installation method, could you please provide the output of those commands:
```shell
# python -c "import fpdf; print(fpdf.__file__)"
# cat $(python -c "import fpdf; print(fpdf.__file__)")
```

**Reply by portfedh** [reply]
Thank you. It worked once I uninstalled  FPDF2 from Anaconda and re installed with pip. 

I tried uninstalling pip and reinstalling anaconda to get the error but this time its working fine, so I guess some dependency did not get uninstalled. 

**Here is the output with pip:**

`$ python -c "import fpdf; print(fpdf.__file__)" `
`>/Users/portfedh/opt/anaconda3/lib/python3.8/site-packages/fpdf/__init__.py`


`$ cat $(python -c "import fpdf; print(fpdf.__file__)")`

```
#!/usr/bin/env python
import sys

from .fpdf import (
    FPDF,
    FPDFException,
    TitleStyle,
    FPDF_FONT_DIR as _FPDF_FONT_DIR,
    FPDF_VERSION as _FPDF_VERSION,
)
from .html import HTMLMixin
from .template import Template
from .deprecation import WarnOnDeprecatedModuleAttributes

FPDF_VERSION = _FPDF_VERSION
"""Current FPDF Version, also available via `__version__` (which is read by `setup.py`):

<pre>
>>> import fpdf
>>> fpdf.__version__
'2.2.0'
</pre>
"""

FPDF_FONT_DIR = _FPDF_FONT_DIR
"""This is the location of where to look for fonts."""

sys.modules[__name__].__class__ = WarnOnDeprecatedModuleAttributes

__license__ = "LGPL 3.0"

__version__ = FPDF_VERSION


__all__ = [
    # metadata
    "__version__",
    "__license__",
    # Classes
    "FPDF",
    "Template",
    "TitleStyle",
    "HTMLMixin",
    # FPDF Constants
    "FPDF_VERSION",
    "FPDF_FONT_DIR",
]
```

Thank you ! I'd been struggling with this for two days....

**Reply by Lucas-C** [reply]
Glad you found a solution!
